// ya-ts-client/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.VersionApi = exports.IdentityApi = exports.GsbApi = exports.NetApi = exports.PaymentApi = exports.ActivityApi = exports.MarketApi = void 0;
exports.MarketApi = __importStar(require("./market-api"));
exports.ActivityApi = __importStar(require("./activity-api"));
exports.PaymentApi = __importStar(require("./payment-api"));
exports.NetApi = __importStar(require("./net-api"));
exports.GsbApi = __importStar(require("./gsb-api"));
exports.IdentityApi = __importStar(require("./identity"));
exports.VersionApi = __importStar(require("./version"));
