// ya-ts-client/dist/activity-api/Client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
const FetchHttpRequest_1 = require("./core/FetchHttpRequest");
const ProviderService_1 = require("./services/ProviderService");
const RequestorControlService_1 = require("./services/RequestorControlService");
const RequestorStateService_1 = require("./services/RequestorStateService");
class Client {
    constructor(config, HttpRequest = FetchHttpRequest_1.FetchHttpRequest) {
        this.request = new HttpRequest({
            BASE: config?.BASE ?? '/activity-api/v1',
            VERSION: config?.VERSION ?? '1.7.0',
            WITH_CREDENTIALS: config?.WITH_CREDENTIALS ?? false,
            CREDENTIALS: config?.CREDENTIALS ?? 'include',
            TOKEN: config?.TOKEN,
            USERNAME: config?.USERNAME,
            PASSWORD: config?.PASSWORD,
            HEADERS: config?.HEADERS,
            ENCODE_PATH: config?.ENCODE_PATH,
        });
        this.provider = new ProviderService_1.ProviderService(this.request);
        this.requestorControl = new RequestorControlService_1.RequestorControlService(this.request);
        this.requestorState = new RequestorStateService_1.RequestorStateService(this.request);
    }
}
exports.Client = Client;
