// ya-ts-client/dist/gsb-api/services/DefaultService.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DefaultService = void 0;
class DefaultService {
    constructor(httpRequest) {
        this.httpRequest = httpRequest;
    }
    /**
     * Upgrades to WebSocket
     *  Upgrades to WebSocket which allows to receive messages to bound GSB services. Check 'gsb-api-messages.yaml' for WebSocket messages specs.
     * @param servicesId Generated id for bound GSB services. It is a base64 encoded GSB address prefix.
     * @returns void
     * @throws ApiError
     */
    getServiceMessages(servicesId) {
        return this.httpRequest.request({
            method: 'GET',
            url: '/services/{servicesId}',
            path: {
                'servicesId': servicesId,
            },
            errors: {
                401: `(401) Authorization information is missing or invalid.`,
                404: `(404) The specified resource was not found.`,
            },
        });
    }
}
exports.DefaultService = DefaultService;
