// ya-ts-client/dist/payment-api/Client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Client = void 0;
const FetchHttpRequest_1 = require("./core/FetchHttpRequest");
const ProviderService_1 = require("./services/ProviderService");
const RequestorService_1 = require("./services/RequestorService");
class Client {
    constructor(config, HttpRequest = FetchHttpRequest_1.FetchHttpRequest) {
        this.request = new HttpRequest({
            BASE: config?.BASE ?? '/payment-api/v1',
            VERSION: config?.VERSION ?? '1.6.3',
            WITH_CREDENTIALS: config?.WITH_CREDENTIALS ?? false,
            CREDENTIALS: config?.CREDENTIALS ?? 'include',
            TOKEN: config?.TOKEN,
            USERNAME: config?.USERNAME,
            PASSWORD: config?.PASSWORD,
            HEADERS: config?.HEADERS,
            ENCODE_PATH: config?.ENCODE_PATH,
        });
        this.provider = new ProviderService_1.ProviderService(this.request);
        this.requestor = new RequestorService_1.RequestorService(this.request);
    }
}
exports.Client = Client;
